dask/array/overlap.py

```diff
@@ -841,7 +843,65 @@ def coerce_boundary(ndim, boundary):
     if isinstance(boundary, tuple):
         boundary = dict(zip(range(ndim), boundary))
     if isinstance(boundary, dict):
-        for i in range(ndim):
-            if i not in boundary:
-                boundary[i] = default
+        boundary = {ax: boundary.get(ax, default) for ax in range(ndim)}
     return boundary
+
+
+def _window_view(arr, window_shape, axis, fix_first_block, block_info):
+    """ core sliding window function"""
+    from .numpy_compat import sliding_window_view
+
+    block_info = block_info[0]
+    slicer = [slice(None, None)] * arr.ndim
+    for ax, window, fix_first in zip(axis, window_shape, fix_first_block):
+        start, stop = None, None
+        is_last_chunk = (
+            block_info["num-chunks"][ax] - 1 - block_info["chunk-location"][ax]
+        ) == 0
+
+        locs = block_info["array-location"][ax]
+        size = locs[1] - locs[0]
+        if not is_last_chunk:
+            stop = -1 * max(window // 2 - 1, 1)
+
+        # when depth == chunk size of first chunk along some axis
+        # we need to make some confusing empirical adjustments.
+        # could be avoided if we rechunked so that min
+        # chunk size is depth+1 (which would also be more efficient)
+        # 1) don't trim the first block along that axis
+        if size == window and block_info["chunk-location"][ax] == 0:
+            stop = None
+        # 2) we need to skip the first "row" of the second block
+        if fix_first and block_info["chunk-location"][ax] == 1:
+            start = 1
+        slicer[ax] = slice(start, stop)
+
+    view = sliding_window_view(arr[tuple(slicer)], window_shape, axis)
+    return view
+
+
+def sliding_window_view(arr, window_shape, axis):
```

Comment:
https://github.com/numpy/numpy/blob/e9070852051515395e0ae9bbe66f929111d59c41/numpy/lib/stride_tricks.py#L117-L123

Does this dispatch/? If so, do we need to move this function to `dask.array.lib.stride_tricks`?

Reply:
@pentschev do you happen to know the answer to this?

Reply:
It does, https://github.com/numpy/numpy/blob/e9070852051515395e0ae9bbe66f929111d59c41/numpy/lib/stride_tricks.py#L122 is what enables that. Moving to `dask.array.lib.stride_tricks` would probably be worth it for that purpose.